core/src/com/unciv/logic/automation/unit/WorkerAutomation.kt

```diff
@@ -648,7 +668,21 @@ class WorkerAutomation(
     private fun getImprovementRanking(tile: Tile, unit: MapUnit, improvementName: String, localUniqueCache: LocalUniqueCache): Float {
         val improvement = ruleSet.tileImprovements[improvementName]!!
         var stats = Stats()
-        if (tile.getOwner() == unit.civ) 
+
+        // Add the value of roads if we want to build it here
+        if (improvement.isRoad() && bestRoadAvailable.improvement(ruleSet) == improvement
+            && tile in tilesOfRoadsToConnectCities) {
+            var value = 1f
+            val city = tilesOfRoadsToConnectCities[tile]!!
+            if (civInfo.stats.statsForNextTurn.gold >= 20)
+                // Bigger cities have a higher priority to connect
+                value += (city.population.population - 3) * .3f
+            // Higher priority if we are closer to connecting the city
+            value += 5 - roadsToConnectCitiesCache[city]!!.size
```

Comment:
So there will be negative ranking if the list is large? Will the worker do anything that's negative? If not they this road will be perpetually unbuilt